src/modules/types.rs

```diff
@@ -1,26 +1,64 @@
 use std::fmt::Display;
 
 use heraclitus_compiler::prelude::*;
+use itertools::Itertools;
 use crate::utils::ParserMetadata;
 
-#[derive(Debug, Clone, PartialEq, Eq, Default)]
+#[derive(Debug, Clone, Eq, Default)]
 pub enum Type {
     #[default] Null,
     Text,
     Bool,
     Num,
+    Union(Vec<Type>),
     Array(Box<Type>),
     Failable(Box<Type>),
     Generic
 }
 
+impl Type {
+    fn eq_union_normal(one: &[Type], other: &Type) -> bool {
+        one.iter().any(|x| (*x).to_string() == other.to_string())
+    }
+
+    fn eq_unions(one: &[Type], other: &[Type]) -> bool {
+        let bigger;
+        let smaller = if one.len() < other.len() { bigger = other; one } else { bigger = one; other };
```

Comment:
I think that this is much easier to read

```suggestion
        let (smaller, bigger) = if one.len() < other.len() {
            (one, other)
        } else {
            (other, one)
        };
```